src/compiler/utilities.ts

```diff
@@ -8194,131 +8198,8 @@ export function getLeftmostExpression(node: Expression, stopAtCallExpressions: b
-    this.flags = flags;
-    if (Debug.isDebugging || tracing) {
-        this.checker = checker;
-    }
-}
-
-function Signature(this: Signature, checker: TypeChecker, flags: SignatureFlags) {
-    // Note: if modifying this, be sure to update SignatureObject in src/services/services.ts
-    this.flags = flags;
-    if (Debug.isDebugging) {
-        this.checker = checker;
-    }
-}
-
-function Node(this: Mutable<Node>, kind: SyntaxKind, pos: number, end: number) {
-    // Note: if modifying this, be sure to update NodeObject in src/services/services.ts
-    this.pos = pos;
-    this.end = end;
-    this.kind = kind;
-    this.id = 0;
-    this.flags = NodeFlags.None;
-    this.modifierFlagsCache = ModifierFlags.None;
-    this.transformFlags = TransformFlags.None;
-    this.parent = undefined!;
-    this.original = undefined;
-    this.emitNode = undefined;
-}
-
-function Token(this: Mutable<Node>, kind: SyntaxKind, pos: number, end: number) {
-    // Note: if modifying this, be sure to update TokenOrIdentifierObject in src/services/services.ts
-    this.pos = pos;
-    this.end = end;
-    this.kind = kind;
-    this.id = 0;
-    this.flags = NodeFlags.None;
-    this.transformFlags = TransformFlags.None;
-    this.parent = undefined!;
-    this.emitNode = undefined;
-}
-
-function Identifier(this: Mutable<Node>, kind: SyntaxKind, pos: number, end: number) {
-    // Note: if modifying this, be sure to update TokenOrIdentifierObject in src/services/services.ts
-    this.pos = pos;
-    this.end = end;
-    this.kind = kind;
-    this.id = 0;
-    this.flags = NodeFlags.None;
-    this.transformFlags = TransformFlags.None;
-    this.parent = undefined!;
-    this.original = undefined;
-    this.emitNode = undefined;
-}
-
-function SourceMapSource(this: SourceMapSource, fileName: string, text: string, skipTrivia?: (pos: number) => number) {
-    // Note: if modifying this, be sure to update SourceMapSourceObject in src/services/services.ts
-    this.fileName = fileName;
-    this.text = text;
-    this.skipTrivia = skipTrivia || (pos => pos);
-}
-
-/** @internal */
-export const objectAllocator: ObjectAllocator = {
-    getNodeConstructor: () => Node as any,
-    getTokenConstructor: () => Token as any,
-    getIdentifierConstructor: () => Identifier as any,
-    getPrivateIdentifierConstructor: () => Node as any,
-    getSourceFileConstructor: () => Node as any,
-    getSymbolConstructor: () => Symbol as any,
-    getTypeConstructor: () => Type as any,
-    getSignatureConstructor: () => Signature as any,
-    getSourceMapSourceConstructor: () => SourceMapSource as any,
-};
-
-const objectAllocatorPatchers: ((objectAllocator: ObjectAllocator) => void)[] = [];
-
-/**
- * Used by `deprecatedCompat` to patch the object allocator to apply deprecations.
- * @internal
- */
-export function addObjectAllocatorPatcher(fn: (objectAllocator: ObjectAllocator) => void) {
-    objectAllocatorPatchers.push(fn);
-    fn(objectAllocator);

```

Comment:
What mechanism are we going to use in the future to tack stuff onto our objects for deprecations? Just throwing things onto the prototype? (That certainly sounds cleaner, though I just removed the one example...)

Reply:
Patching the prototype seems to cause memory usage to skyrocket. I'd maybe do what I'm doing with `SymbolObjectInternals`, such that using the method throws in TSC, but the deprecation code could patch that object with a version that works (with warnings).

Reply:
Makes sense; just didn't want to accidentally close the door and I did remove the one test we had for that perf which made it harder for you. 🙁